### PR TITLE
feat: support for React 18

### PR DESCRIPTION
**Pull request by huksley**
Adds react 18 to peer dependencies

### CODE REVIEW

**Comment by huksley** [comment]
Published as @huksley/react-responsive-modal@6.3.3

**Comment by Maciej916** [comment]
I installed @huksley/react-responsive-modal@6.3.3 and it seems like flicker effect still exist, i don't know if it is only for me because im using nextjs

**Comment by huksley** [comment]
I haven't fixed the flicker in this PR.

In my code which uses `react-responsive-modal` I am doing this fix https://github.com/pradel/react-responsive-modal/issues/495#issuecomment-1174789792

**Comment by Maciej916** [comment]
> I haven't fixed the flicker in this PR.
> 
> In my code which uses `react-responsive-modal` I am doing this fix [#495 (comment)](https://github.com/pradel/react-responsive-modal/issues/495#issuecomment-1174789792)

I thought you implemented this because of your comment but it, my mistake, but it would be good if this is implemented in the release so there is no need to repeat this code each modal.

**Comment by huksley** [comment]
It seems that React 18 support fixed in [1.3.1](https://www.npmjs.com/package/@bedrock-layout/use-forwarded-ref/v/1.3.1) of `@bedrock-layout/use-forwarded-ref`

File: react-responsive-modal/package.json [hunk 1]
```diff
@@ -55,13 +55,12 @@
     }
   ],
   "dependencies": {
-    "@bedrock-layout/use-forwarded-ref": "^1.1.4",
```
Comment (pradel) [inline react-responsive-modal/package.json:58]: now that the latest version support react 18 could you please just upgrade the lib?

Comment (huksley) [reply react-responsive-modal/package.json:58]: done

File: react-responsive-modal/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-responsive-modal",
-  "version": "6.2.0",
+  "version": "6.3.2",
```
[the remark this answers is not in the document]
Comment (huksley) [reply react-responsive-modal/package.json]: done

**Comment by ashtonpaul** [comment]
@pradel  Anything preventing this from being merged upstream to npm? Thanks in advance.

**Comment by ArturoDucasse** [comment]
Any updates on this PR? @pradel

**Comment by pradel** [comment]
Published in 6.4.1, thanks for the pr!

**Pull request merged**